OTAnalytics/domain/filter.py

```diff
@@ -149,10 +167,50 @@ def build_filter(self, filter_builder: FilterBuilder) -> Filter:
         if self.classifications:
             filter_builder.add_has_classifications_predicate(self.classifications)
 
-        if self.start_date and self.end_date:
-            filter_builder.add_starts_at_or_after_date_predicate(self.start_date)
+        if self.date_range.start_date and self.date_range.end_date:
+            filter_builder.add_starts_at_or_after_date_predicate(
+                self.date_range.start_date
+            )
+
+        if self.date_range.end_date:
+            filter_builder.add_ends_before_or_at_date_predicate(
+                self.date_range.end_date
+            )
+
+        filter_builder.build()
 
-        if self.end_date:
-            filter_builder.add_ends_before_or_at_date_predicate(self.end_date)
+        return filter_builder.get_result()
 
-        return filter_builder.build()
+
+class FilterElementSettingRestorer:
+    def __init__(self) -> None:
+        self._by_date_filter_setting: Optional[DateRange] = None
+        self._by_classification_filter_setting: Optional[list[str]] = None
+
+    def save_by_date_filter_setting(self, filter_element: FilterElement) -> None:
+        self._by_date_filter_setting = filter_element.date_range
+
+    def save_by_classification_filter_setting(
+        self, filter_element: FilterElement
+    ) -> None:
+        self._by_classification_filter_setting = filter_element.classifications.copy()
```

Comment:
Actually I was just taking precautions for unwanted mutations of the list.
The reason why I didn't do the same for the date filter was because of `date_range` being an immutable/frozen dataclass. But I think it's better to return a copy to avoid confusion